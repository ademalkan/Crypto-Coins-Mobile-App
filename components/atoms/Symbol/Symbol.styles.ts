import { StyleSheet } from "react-native";

export default StyleSheet.create({
  symbol: {
    fontSize: 14,
    color: '#666',
    textTransform: "uppercase"
  },
})