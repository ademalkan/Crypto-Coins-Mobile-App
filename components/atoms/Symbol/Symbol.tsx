import React from 'react';
import { Text } from 'react-native';
import styles from "./Symbol.styles";

type SymbolProps = {
  symbol: string;
};

const Symbol: React.FC<SymbolProps> = ({ symbol }) => {
  return (
    <Text style={styles.symbol}>{symbol}</Text>
  );
};

export default Symbol;
