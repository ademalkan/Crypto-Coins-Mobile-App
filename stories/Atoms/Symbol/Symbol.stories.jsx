import React from "react";
import { View } from "react-native";
import Symbol from "../../../components/atoms/Symbol/Symbol";

export default {
  title: "Atoms/Symbol",
  component: Symbol,
  decorators: [
    (Story) => (
      <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
        <Story />
      </View>
    ),
  ],
};

const symbol = "BTC";

export const Default = () => <Symbol symbol={symbol} />;
