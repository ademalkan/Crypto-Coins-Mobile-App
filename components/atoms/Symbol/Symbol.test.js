import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import Symbol from "./Symbol";

describe("Symbol", () => {
  it("should render the symbol prop correctly", () => {
    const symbol = "BTC";

    const component = renderer.create(<Symbol symbol={symbol} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
