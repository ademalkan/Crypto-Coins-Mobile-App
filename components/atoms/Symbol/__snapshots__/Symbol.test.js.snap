// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Symbol should render the symbol prop correctly 1`] = `
<Text
  style={
    {
      "color": "#666",
      "fontSize": 14,
      "textTransform": "uppercase",
    }
  }
>
  BTC
</Text>
`;
